dev/storage_base.h

```diff
@@ -885,15 +924,40 @@ namespace sqlite_orm {
                 return result;
             }
 
-            void drop_table_internal(sqlite3* db, const std::string& tableName) {
+            void drop_table_internal(sqlite3* db, const std::string& tableName, bool ifExists) {
                 std::stringstream ss;
-                ss << "DROP TABLE " << streaming_identifier(tableName) << std::flush;
+                ss << "DROP TABLE";
+                if(ifExists) {
+                    ss << " IF EXISTS";
+                }
+                ss << streaming_identifier(tableName) << std::flush;
                 perform_void_exec(db, ss.str());
             }
 
-            static int collate_callback(void* arg, int leftLen, const void* lhs, int rightLen, const void* rhs) {
-                auto& f = *(collating_function*)arg;
-                return f(leftLen, lhs, rightLen, rhs);
+            void drop_index_internal(const std::string& indexName, bool ifExists) {
+                std::stringstream ss;
+                ss << "DROP INDEX";
+                if(ifExists) {
+                    ss << " IF EXISTS";
+                }
+                ss << quote_identifier(indexName) << std::flush;
+                perform_void_exec(this->get_connection().get(), ss.str());
+            }
+
+            void drop_trigger_internal(const std::string& triggerName, bool ifExists) {
+                std::stringstream ss;
+                ss << "DROP TRIGGER";
```

Comment:
The trailing whitespace character is necessary.

Reply:
done